zipline/history/history.py

```diff
@@ -278,6 +285,8 @@ def __init__(self, bar_count, frequency, field, ffill,
         # The frequency at which the data is sampled.
         self.frequency = frequency
         # The field, e.g. 'price', 'volume', etc.
+        if not isinstance(field, Hashable):
```

Comment:
This is only difference if you are a bully and do something like:

```
class C(object):
    @property
    def __hash__(self):
        raise AttributeError()
```

But the abc is cleaner imho